### ISSUE TITLE
`dogstatsd` support

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#15741]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Users want a dogstatsd compatible receiver. The current `statsd` receiver attempts to match the [canonical statsd](https://github.com/statsd/statsd) implementation.

### Proposal

Move `statsd` to a codec and add a `dogstatsd` codec. Both of these could be used with the `socket` source. We could continue to provide a `statsd` source as "syntax sugar" over the `socket` source.

### References

- https://github.com/vectordotdev/vector/issues/15683#issuecomment-1362199941

### Version

v0.26.0

**Comment by zamazan4ik** [comment]
@jszwedko according to the `statsd` source description, it support metrics emitted by `statsd` aggregator, not by `statsd` clients. I am wondering - could I use `statsd` source with Vector as a original Statsd drop-in replacement? If yes, it would be wonderful.

**Comment by zamazan4ik** [comment]
I've just checked - seems like Vector works fine as a Statsd aggregator itself. So I guess we need to clarify a little bit the documentation since right now it's not clear enough (at least for me and a couple of other people).

**Comment by jszwedko** [comment]
> @jszwedko according to the `statsd` source description, it support metrics emitted by `statsd` aggregator, not by `statsd` clients. I am wondering - could I use `statsd` source with Vector as a original Statsd drop-in replacement? If yes, it would be wonderful.

Good callout, I do see that in the docs now. I think the docs likely need to be clarified here. I believe the source generally works with statsd clients as well (and should).

**Comment by zamazan4ik** [comment]
> I believe the source generally works with statsd clients as well (and should).

Yeah. I have tested it at least with Hashicorp Vault with enabled Statsd integration on it and a Vector instance with Statsd source. I will create the corresponding PR with the updated documentation.

Thanks for the help!

**Comment by jszwedko** [comment]
Newer agents also support receiving multiple datapoints in a single line: https://github.com/DataDog/datadog-agent/tree/20bcaf3df7b9240e14c6ec0394cf582b4b6c5ea2/comp/dogstatsd
